File: 프로그래머스/2/12902. 3 x n 타일링/3 x n 타일링.java
class Solution {
    public int solution(int n) {
        
        long[] dp = new long[n+1];
        long sum = 0;
        
        dp[2] = 3;
        
        for(int i = 4; i < n+1; i+=2){
            dp[i] = dp[i-2] *3 + sum*2 + 2;
            dp[i] %= 1000000007;
            sum += dp[i-2];
        }
        
        return (int)dp[n];
    }
}

File: 프로그래머스/2/12902. 3 x n 타일링/3 x n 타일링.js
function solution(n) {
    
    const dp = Array(n+1).fill(0);
    
    dp[2] = 3;
    var sum = 0;

    for(let i=4; i<n+1; i+=2){
        dp[i] = dp[i-2]*3 + sum*2 + 2;
        if(dp[i] > 1000000007) dp[i] %= 1000000007;
        sum += dp[i-2];
    }
    
    return dp[n];
}